app/middleware/Redirect.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace GanttDashboard\App\Middleware;
+
+use Phalcon\Application;
+use Phalcon\Events\Event;
+use GanttDashboard\App\Services\Authentication as AuthenticationService;
+
+/**
+ * Redirect
+ *
+ * Checks the request and redirects the user somewhere else if needed
+ */
+class Redirect
+{
+    /**
+     * @var AuthenticationService
+     */
+    private $authenticationService;
+
+    /**
+     * Constructs the needed services, set in DI, for authentication service
+     * @param AuthenticationService $authenticationService
+     */
+    public function __construct(
+        AuthenticationService $authenticationService
+    ) {
+        $this->authenticationService = $authenticationService;
+    }
+
+    /**
+     * Before anything happens
+     *
+     * @param Event $event
+     * @param Application $application
+     *
+     * @return bool
+     */
+    public function beforeSendResponse(Event $event, Application $application)
+    {
+        $restrictedUri = [
```

Comment:
are you really going to put here all restricted urls?
also try to check ACL and see if that integration with a middleware fits better

Reply:
Replace the event with something else: you need to do all the checks BEFORE you are handling the request in the contrller, not after